lightning/src/util/events.rs

```diff
@@ -505,6 +505,27 @@ pub enum Event {
 		/// transaction.
 		claim_from_onchain_tx: bool,
 	},
+	/// Used to indicate that a previously opened channel with the given `channel_id` is ready to
+	/// be used. This event is emitted either when the funding transaction has been confirmed
+	/// on-chain, or, in case of a 0conf channel, when both parties have confirmed the channel
+	/// establishment.
+	ChannelEstablished {
```

Comment:
Not 100% sure about the name, since `ChannelEstablished` is somewhat confusing since the event should be emitted only once on channel establishment, not on reestablishment, etc. I thought about naming this `ChannelReady` (could also be confusing for people knowing about LN internals) or `ChannelConfirmed` (doesn't really work for 0conf channels..) before. So, open for further suggestions.

Reply:
Maybe `ChannelUsable` is more clear? It feels like that's what users would care about

Reply:
I think from a public API perspective it isn't used anywhere else, except in `usable_channels`, which is a nice alignment. No strong feelings here though

Reply:
Oh, actually I see your point. Maybe it is confusing so feel free to disregard

Reply:
So, given that the new event should be the counterpart to `ChannelClosed`, could we just name it `ChannelOpened`? Might also be a bit in conflict with `ChannelOpenRequest`, but seems the most straightforward?

Reply:
Perhaps `ChannelOperational` could work? We've used the word "operational" for some error messages when receiving htlc messages too early.

Reply:
Went with `ChannelOpened` for now. Let me know if there anyone has strong feelings against that.